staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -431,10 +559,32 @@ func NewDiffer(from, to string) (*Differ, error) {
 
 // Diff diffs to versions of a specific object, and print both versions to directories.
 func (d *Differ) Diff(obj Object, printer Printer) error {
-	if err := d.From.Print(obj, printer); err != nil {
+	from, err := d.From.getObject(obj)
+	if err != nil {
+		return err
+	}
+	to, err := d.To.getObject(obj)
+	if err != nil {
+		return err
+	}
+
+	// Mask secret values if object is V1Secret
+	o, err := obj.Merged()
+	if err != nil {
+		return err
+	}
```

Comment:
Do you need to get "Merged" just to check the gvk? I think you should be able to do it either from "From" or "To".

Reply:
You're right, fixed!